bin/agitator

```diff
@@ -30,6 +30,144 @@ Possible commands:
 EOF
 }
 
+function run_agitators() {
+  unset APP_HOSTS
+  export APP_HOSTS=$MANAGER_HOSTS
+  echo "Starting manager agitation as ${1} for ${APP_HOSTS}"
+  start_app_agitator "manager" "${AGTR_MASTER_KILL_SLEEP_TIME}" "${AGTR_MASTER_RESTART_SLEEP_TIME}" 1 1 > "${manager_log}" 2>&1 &
+  unset APP_HOSTS
+  export APP_HOSTS=$TSERVER_HOSTS
+  echo "Starting tserver agitation as ${1} for ${APP_HOSTS}"
+  start_app_agitator "tserver" "${AGTR_TSERVER_KILL_SLEEP_TIME}" "${AGTR_TSERVER_RESTART_SLEEP_TIME}" "${AGTR_TSERVER_MIN_KILL}" "${AGTR_TSERVER_MAX_KILL}" > "${tserver_log}" 2>&1 &
+}
+
+# Starts a app specific agitator
+# usage: start_app_agitator app_name kill_sleep_time restart_sleep_time min_kill max_kill
+# Requires that a list of hosts to be set in $APP_HOSTS
+function start_app_agitator() {
+  app_name=$1
+  kill_sleep_time=$2
+  restart_sleep_time=$3
+  min_kill=$4
+  max_kill=$5
+  ENV_VARS="ACCUMULO_HOME=${ACCUMULO_HOME} ZOOKEEPER_HOME=${ZOOKEEPER_HOME} HADOOP_HOME=${HADOOP_HOME} JAVA_HOME=${JAVA_HOME}"
+  if [[ -z $APP_HOSTS ]]; then
+    echo "ERROR: No hosts were found in env for ${app_name}"
+    exit 1
+  fi
+  T="$(date +'%Y%m%d %H:%M:%S')"
+  echo "${T} Running ${app_name} agitator ${kill_sleep_time} ${restart_sleep_time} ${min_kill} ${max_kill} for HOSTS=${APP_HOSTS}"
+  while true; do
+    echo "${T} Sleeping for ${kill_sleep_time} minutes"
+    sleep $((kill_sleep_time * 60))
+
+    T="$(date +'%Y%m%d %H:%M:%S')"
+    numToWack=$((min_kill + RANDOM % max_kill))
+    nodeToWack=$(echo "${APP_HOSTS}" | cut -d" " -f1)
+    hostsArray=("${APP_HOSTS}")
+    numOfHosts=${#hostsArray[@]}
+    declare -a nodesToWack
+
+    if ((max_kill == 1)) ; then
+      echo "${T} Killing ${app_name}"
+      ssh "${nodeToWack}" "pkill -f '[ ]org.apache.accumulo.start.*${app_name}'"
+    else
+      # get the random nodes to kill
+      count=0
+      while [ $count -lt $numToWack ]; do
```

Comment:
Should use `[[ ]]` instead of `[ ]` to avoid any possible performance hit by calling `/usr/bin/[`, or any possible confusion because it looks like that's what it's doing. `[[ ]]` is built-in, but `[ ]` is a standalone executable that bash may or may not execute or replace with its own built-in interpretation (just like /usr/bin/echo).